### DISCUSSION TITLE
No "process complete" message after msi installation

### DISCUSSION THREAD

**Discussion by FearlessPixel** [wiverson/maven-jpackage-template#37] [Q&A] [answered]
Hi,

I would like to add some kind of completion message to the msi installer but I'm having a hard time figuring it out. Can this be added to the "add-launch-to-msi.js" file? I would really appreciate a solution or if someone can point me to the API I'm supposed to be using for that file.

**Comment by wiverson** [comment, marked as answer]
The installer packaging for an msi is driven by Wix and Windows installer. The add-launch-to-msi.js file is a script that tweaks the Windows installer file. Windows installer is a big old complicated thing. I was able to work out the details to create that file by a combination of combing through Reddit and Stack Overflow.

Here are a few resources:

https://wixtoolset.org/
[WiX 3.6: A Developer's Guide to Windows Installer XML](https://amzn.to/39QydMI)

In theory you can do anything, up to and including crazy stuff like embedding other executables. Certainly a completion notification is possible.

IIRC there is a way to tell the msi to run with different display levels - https://www.advancedinstaller.com/user-guide/msiexec.html - but I don't recall right off the commands to set it.

One complicating factor is that most system administrators doing rollouts specifically don't want any kind of completion dialog, whereas a lot of users like the option. Just depends on your scenario.
